transport/transport.go

```diff
@@ -337,14 +356,10 @@ func (t *Transport) dial(retryCtx context.Context, remote id.Signatory, remoteAd
 			case <-retryCtx.Done():
 			case <-dialCtx.Done():
 				if !t.IsConnected(remote) {
-					// Cancel current dial context if restarting loop
-					cancel()
 					continue
 				}
 			}
 		}
-		// Cancel last dial context before exiting
-		cancel()
```

Comment:
Why was this removed?

Reply:
I was removing the `defer cancel()` in the loop and replacing it with specific cancel calls